### PR TITLE
Integrate gem push in Jenkins pipeline

### PR DESCRIPTION
**Pull request by ciremoussadia**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -67,6 +67,27 @@ pipeline {
...
       }
     }
+
+    stage ('Publish to rubygems.org') {
+      environment {
+        GEM_HOST_API_KEY = credentials('jenkins-rubygems-api-key')
+      }
+      when {
+        anyOf {
+          branch 'master'
```
Comment (blambeau) [inline Jenkinsfile:77]: Not on master.

File: Jenkinsfile [hunk 2]
```diff
@@ -67,6 +67,27 @@ pipeline {
...
+          buildingTag()
+        }
+        steps {
+          container('builder') {
+            script {
+              withCredentials([]) {
```
Comment (blambeau) [inline Jenkinsfile]: Sure about that one? @llambeau maybe?

File: Makefile [hunk 3]
```diff
@@ -64,3 +64,14 @@ up: Dockerfile.built
...
+################################################################################
+### Gem Management
+###
+GEM = $(shell find . -name "talktome*.gem" )
+
+gem.built:
+	gem build talktome.gemspec
+
+gem.publish: gem.built
+	gem push $(GEM) --key GEM_HOST_API_KEY
```
Comment (blambeau) [inline Makefile:77]: I'm not the `--key` is necessary

**Review by blambeau — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (ciremoussadia) [reply Makefile:77]: The --key is used to pass the token

File: Jenkinsfile (hunk 1 above)
Comment (ciremoussadia) [reply Jenkinsfile:77]: @blambeau ok so when do we want it to be triggered ?

File: Jenkinsfile (hunk 2 above)
Comment (ciremoussadia) [reply Jenkinsfile]: yeah of course this should not be herre.

File: .gitignore [hunk 4]
```diff
@@ -10,3 +10,4 @@ Gemfile.lock
 /Dockerfile.log
 /Dockerfile.pushed
 /pkg
+talktome.gem
```
Comment (blambeau) [inline .gitignore:13]: Remove (see later)

File: Jenkinsfile [hunk 5]
```diff
@@ -12,13 +12,32 @@ pipeline {
...
+          }
+        }
+      }
+    }
+
+    stage ('Building Talktome Gem') {
+      steps {
+        container('builder') {
+          script {
+            sh 'make talktome.gem'
```
Comment (blambeau) [inline Jenkinsfile:35]: Let's call it `make gem` instead.

File: Makefile [hunk 6]
```diff
@@ -26,8 +26,9 @@ $(info $(TINY) $(MINOR) $(MAJOR))
 clean:
 	rm -rf pkg/
 	rm -rf Dockerfile.log Dockerfile.built Dockerfile.pushed
+	rm -rf talktome.gem
```
Comment (blambeau) [inline Makefile:29]: `rm -rf pkg/*`

File: Makefile [hunk 7]
```diff
@@ -26,8 +26,9 @@ $(info $(TINY) $(MINOR) $(MAJOR))
...
 
-Dockerfile.built: Dockerfile $(shell git ls-files)
+Dockerfile.built: Dockerfile talktome.gem $(shell git ls-files)
```
Comment (blambeau) [inline Makefile:31]: I dont think it's needed.

File: Makefile [hunk 8]
```diff
@@ -62,5 +63,15 @@ Gemfile.lock: Gemfile
...
-	docker run --rm -e TALKTOME_EMAIL_DEFAULT_FROM=from@talktome.com $(IMAGE) bundle exec rake test
+test: Dockerfile.built
+	docker run --rm -e TALKTOME_EMAIL_DEFAULT_FROM=from@talktome.com $(IMAGE) sh -c "bundle install && bundle exec rake test"
+
+################################################################################
+### Gem Management
+###
+
+talktome.gem:
+	docker run -t --rm -v ${PWD}/:/app -w /app ruby bash -c "gem build talktome.gemspec -o talktome.gem"
```
Comment (blambeau) [inline Makefile:74]: There is a `bundle exec rake gem` available on all projects, that puts the gem in `pkg/...`. We need to use that one.

File: Makefile [hunk 9]
```diff
@@ -62,5 +63,15 @@ Gemfile.lock: Gemfile
...
+
+gem.publish: talktome.gem
+	docker run -t --rm -v ${PWD}/:/app -w /app -e GEM_HOST_API_KEY=${GEM_HOST_API_KEY} ruby bash -c "gem push talktome.gem"
```
Comment (blambeau) [inline Makefile:77]: And this one must probably be adapted, therefore.

**Review by blambeau — CHANGES_REQUESTED** [review]

**Comment by OSdave** [comment]
@ciremoussadia and @blambeau the build passes, please review the latest version

**Review by blambeau — APPROVED** [review]
@OSdave Looks good to me. Lets squash, rebase, and merge. And see what happens if you tag `1.2.0`.

**Pull request merged**